tests-arquillian/src/test/java/org/jboss/weld/tests/extensions/staticObserver/StaticLifecycleEventObserverTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.weld.tests.extensions.staticObserver;
+
+import javax.enterprise.inject.spi.Extension;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.ShouldThrowException;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.BeanArchive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.weld.test.util.Utils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Testing that a deployment error occurs extension has static lifecycle event observer method
+ *
+ * @author Antoine Sabot-Durand
+ */
+@RunWith(Arquillian.class)
+public class StaticLifecycleEventObserverTest {
+    @Deployment
+    @ShouldThrowException(StaticLifecycleEventObserverForTestException.class)
```

Comment:
I don't think so. The "misleading" exception is `IllegalStateException` whereas the new one is `DefinitionException`. Anyway, we don't test exception messages because it's not possible to create a reliable `DeploymentExceptionTransformer` for all container adapters. Even `WildFly8DeploymentExceptionTransformer` contains some hacks just to support `DefinitionException` and `DeploymentException`.